### PR TITLE
Henri/schema

### PR DESCRIPTION
**Pull request by Vanderscycle**
schemaBuilder is a building block that allows to create a presentation request and correctly make that api call to the backend. Next step is to replace the mockend db with calls to mongodb

### CODE REVIEW

File: frontend/src/api/presentationAxios.ts [hunk 1]
```diff
@@ -14,3 +15,16 @@
     }
   }
 
+  export async function postNewPresentationRequest(payload:PostPresentationPayload,endpoint:string=`${baseUrl}/presentations`):Promise<any> {
+  try {
+    console.log(endpoint)
+    const response = await axios.post(endpoint, payload)
+    if(response.status == 201){
+        // test for status you want, etc
+        console.log(response.status)
```
Comment (brianorwhatever) [inline frontend/src/api/presentationAxios.ts]: this should probably return the data inside of this or else 400's will blow up

File: frontend/src/routes/Presentations.svelte [hunk 2]
```diff
@@ -1,69 +1,37 @@
...
-      console.log(res)
-      return res
+  async function apiGetCall():Promise<any>{
+    let res = await getPresentations()
+    console.log(res)
+    return res
   }
   let displayData = false
-  let data:PresentationTableInterface[] = []
+  let data:Array<string> = []
```
Comment (brianorwhatever) [inline frontend/src/routes/Presentations.svelte]: this shouldn't be an Array type it should be a list of string type
```suggestion
  let data: string[] = []
```

Comment (brianorwhatever) [reply frontend/src/routes/Presentations.svelte]: and all other similar instances

File: frontend/src/api/presentationAxios.ts (hunk 1 above)
Comment (Vanderscycle) [reply frontend/src/api/presentationAxios.ts]: good point, I've changed the return of both axios post and get functions

File: frontend/src/routes/Presentations.svelte (hunk 2 above)
Comment (Vanderscycle) [reply frontend/src/routes/Presentations.svelte]: sounds good, I'll pay more attention to Array<t> that I infer. I often use them interchangeably. Made changes

File: frontend/src/interfaces.ts [hunk 3]
```diff
@@ -5,26 +5,48 @@ export interface routeHooks {
...
+export interface NameSchema {
+  name: string;
+  schema:string;
+  store:Writable<any>|undefined;
+}
+//api
+export interface PostPresentationPayload {
+  name: string;
+  schema:string;
+  paths: Array<any>;
```
Comment (brianorwhatever) [inline frontend/src/interfaces.ts]:
```suggestion
  paths: string[];
```

Comment (Vanderscycle) [reply frontend/src/interfaces.ts]: modified to std

**Pull request merged**